packages/clay-data-provider/src/useResource.ts

```diff
@@ -0,0 +1,236 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {
+	FetchPolicy,
+	IDataProvider,
+	NetworkStatus,
+	SYMBOL_ORIGIN,
+	TVariables,
+} from './types';
+import {timeout} from './util';
+import {useCache} from './useCache';
+import {useDebounce} from '@clayui/shared';
+import {useEffect, useRef, useState} from 'react';
+
+interface IResource extends IDataProvider {
+	onNetworkStatusChange?: (status: NetworkStatus) => void;
+}
+
+const getQueryParams = (variables: any): string | null => {
+	if (!variables) {
+		return null;
+	}
+
+	const keys = Object.keys(variables);
+
+	return keys
+		.map((key, index) => {
+			const param = `${key}=${variables[key]}`;
```

Comment:
Do we need to escape `variables[key]`? Thinking this might lead to xss issues.

Reply:
Hm yeah, I'll go ahead and use `metal-uri` since it handles some things, so it will be easier than implementing some logic here.